extensions/kafka-streams/deployment/src/main/java/io/quarkus/kafka/streams/deployment/KafkaStreamsProcessor.java

```diff
@@ -56,7 +56,19 @@ void build(RecorderContext recorder,
         reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, false, ByteArraySerde.class));
         reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, false, FailOnInvalidTimestamp.class));
 
-        nativeLibs.produce(new SubstrateResourceBuildItem(Environment.getJniLibraryFileName("rocksdb")));
+        // for RocksDB, either add linux64 native lib when targeting containers
+        String containerRuntime = System.getProperty("native-image.container-runtime");
+        if (containerRuntime != null) {
+            containerRuntime = containerRuntime.trim();
+
+            if (containerRuntime.equals("docker") || containerRuntime.equals("podman")) {
+                nativeLibs.produce(new SubstrateResourceBuildItem("librocksdbjni-linux64.so"));
```

Comment:
For my own understanding, is it included in the kafka client?

Reply:
It's included in a dependency of Streams, _rocksdbjni-5.14.2.jar_. RocksDB will extract the one for the current platform from its own JAR (or the application image in our native case) into a temp dir and load it from there via JNI. I could simply add all the native libs, but for the sake of smaller size I tried to just add the one that's really needed for the native target.